### PR TITLE
152 implement api endpoint for retrieving user input history

### LINKED ISSUE #152: Implement API Endpoint for Retrieving User Input History
**Related User Story**
#120 

**Related Tickets**
#148

**Description**
I need to create a new API endpoint that allows the retrieval of all related user input history. This endpoint will be consumed by the frontend to display the input history to the user.

**Acceptance Criteria**
- Create a new route in the backend API to handle the retrieval of all related UserInput.
- Implement the logic to retrieve all related UserInput from the database.
- Return the retrieved UserInput data as a response from the API.

### PR DESCRIPTION
**Pull request by kinskiwu**
## Description
- Created a new route in the backend API to handle the retrieval of all related UserInput.
- Implemented the logic to retrieve all related UserInput from the database & return the retrieved UserInput data as `{"userInputs": [array of UserInputs]}` from the API.
- Added a `/mocks` folder to store mock data.
- Added NODE_ENV to scripts for consistent environment settings.

Fixes #152 

## Type of change
- [X] ✨  New feature (non-breaking change which adds functionality)
- [X] 📝  Documentation update

## How Has This Been Tested?

- [X] Manual testing with thunder client
image

## Checklist:

- [X] I have performed a self-review of my own code.
- [X] I have commented on my code, particularly in hard-to-understand areas.
- [X] My changes generate no new warnings.
- [X] I have made corresponding changes to the documentation.
- [X] I have reviewed the [**Security Best Practices Document**](./docs/SECURITY_BEST_PRACTICES.md).
- [X] I have followed the [**Naming Conventions Guide**](./docs/NAMING_CONVENTIONS_GUIDE.md) for my changes.
- [X] I have ensured that my pull request title is descriptive.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,15 +4,15 @@
   "description": "A learning tool for people to learn fast and easy",
   "main": "index.js",
   "scripts": {
-    "dev": "concurrently \"npm run dev:frontend\" \"npm run dev:backend\"",
-    "dev:frontend": "vite --config client/vite.config.js",
-    "dev:backend": "nodemon server/server.ts",
+    "dev": "NODE_ENV=development concurrently \"npm run dev:frontend\" \"npm run dev:backend\"",
+    "dev:frontend": "vNODE_ENV=development vite --config client/vite.config.js",
```
Comment (relandboyle) [inline package.json:8]: Is that little leading 'v' a typo?

**Review by relandboyle — APPROVED** [review]
So fast!

File: package.json (hunk 1 above)
Comment (kinskiwu) [reply package.json]: Good catch! Just removed the extra `v`.

**Review by ChTiSh — APPROVED** [review]
Thank you so much for the mock data folder @AaronDreyfuss

**Pull request merged**